src/Compilers/Test/Core/CompilationVerifier.cs

```diff
@@ -446,5 +446,25 @@ public void VerifySynthesizedFields(string containingTypeName, params string[] e
                 .ToList();
             AssertEx.SetEqual(expectedFields, members);
         }
+
+        /// <summary>
+        /// Useful for verifying the expected variables are hoisted for closures, async, and iterator methods.
+        /// Additionally this overload performs validations against the produced assembly.
+        /// </summary>
+        public void VerifySynthesizedFields(string containingTypeName, Func<IModuleSymbol, INamedTypeSymbol> findContainer, int arity, params string[] expectedFields)
+        {
+            VerifySynthesizedFields(containingTypeName, expectedFields);
+
+            CommonTestBase.RunValidators(this, assemblyValidator: null, symbolValidator: module =>
+            {
+                var container = findContainer(module);
+                AssertEx.NotNull(container);
+                Assert.Equal(arity, container.Arity);
+
+                var fields = container.GetMembers().OfType<IFieldSymbol>().Select(field => $"{field.Type} {field.Name}").ToArray();
```

Comment:
>field.Type [](http://example.com/codeflow?start=94&length=10)

To convert types to strings for test purposes, we usually use test display mode. See usages of ```ToTestDisplayString``` helper.

Reply:
Sure, changed in part of eecb17b.